simulator/base_simulator.py

```diff
@@ -1,31 +1,19 @@
 import numpy as np
-from pynput import keyboard
 import time
+from abc import ABC, abstractmethod
+from pynput import keyboard
 
-class GridworldSimulator:
+class Simulator(ABC):
     def __init__(self, environment, initial_state):
-        self.exitProgram = 0
-
         self.environment = environment
-        self.grid = self.environment.grid
         self.state = initial_state
 
-    def run(self):
-        '''
-        Blocking function to run the simulator
-        '''
-        # keyboard listener in background thread
-        listener = keyboard.Listener(on_press=self.takeAction)
-        listener.start()
-
-        self.drawGridworld()
-        while not self.exitProgram:
-            time.sleep(0.2)
+        self.exitProgram = 0
     
-    def moveToNextState(self, action):
+    def nextStep(self, action):
```

Comment:
We may want to further abstract this for the continuous case. It doesn't make sense to talk about P for a continuous state since the probability of any given value will be zero.